src/Microsoft.NetCore.Analyzers/UnitTests/Security/UseDefaultDllImportSearchPathsAttributeTests.cs

```diff
@@ -0,0 +1,178 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using Microsoft.CodeAnalysis.Diagnostics;
+using Test.Utilities;
+using Xunit;
+
+namespace Microsoft.NetCore.Analyzers.Security.UnitTests
+{
+    // All the test cases use user32.dll as an example,
+    // however it is a commonly used system dll and will be influenced by Known Dlls mechanism,
+    // which will ignore all the configuration about the search algorithm.
+    // Fow now, this rule didn't take Known Dlls into consideration.
+    // If it is needed in the future, we can recover this rule.
+    public class UseDefaultDllImportSearchPathsAttributeTests : DiagnosticAnalyzerTestBase
+    {
+        // It will try to retrieve the MessageBox from user32.dll, which will be searched in a default order:
+        //  1. The directory from which the application is loaded.
+        //  2. The current directory.
+        //  3. The system directory, usually C:\\Windows\\System32\\ (The GetSystemDirectory function is called to obtain this directory.).
+        //  4. The 16-bit system directory – There is no dedicated function to retrieve the path of this directory, but it is searched as well.
+        //  5. The Windows directory. The GetWindowsDirectory function is called to obtain this directory.
+        //  6. The directories that are listed in the PATH environment variable
```

Comment:
I don't think we should try to document behavior here, since it varies a lot, we're unlikely to be accurate.